### ISSUE TITLE
 The filer can not fetch file from volume after Upgrade from 2.17 to 2.21. 

### ISSUE THREAD

**Issue by chaos369** [chrislusf/seaweedfs#1860]
**Describe the bug**
 The filer can not fetch file from volume after Upgrade from 2.17 to 2.21. 

**System Setup**
nohup ./weed -logdir=/root/weed/log/master  master -ip.bind=127.0.0.1 -ip=127.0.0.1 &

nohup ./weed -logdir=/root/weed/log/vol2 volume -dir=/mnt/sdb/data/weed -max=0 -mserver=127.0.0.1:3333 -port=9090 -ip=[2a01:438:191:60d3::8]  -ip.bind=[2a01:438:191:60d3::8]&
nohup ./weed -logdir=/root/weed/log/vol2 volume -dir=/mnt/sdb/data/weed -max=0 -mserver=127.0.0.1:3333 -port=9090 -ip=[2a01:438:191:60d3::9]  -ip.bind=[2a01:438:191:60d3::9]&

nohup ./weed -logdir=/root/weed/log/filer  filer  -collection="data" -master=localhost:3333 -ip.bind=127.0.0.1 -ip=127.0.0.1 -port=5555 &

 package: linux_amd64.tar.gz
 OS: ubuntu 20.04
 weed: 2.21
filer.toml file: no 

**Expected behavior**
It's OK in 2.17 for the same configure.

**Screenshots**

I0305 05:02:30 06188 filechunk_manifest.go:112] read http://[2a01:438:191:60d3::8]:9090/35,e41340498578 failed, err: couldn't find DNS entries for the given domain. Try using DialDualStack

**Comment by chrislusf** [comment]
This has been fixed already in https://github.com/chrislusf/seaweedfs/commit/e982b06bcdbb1d5f5433d4befd1b457e0bc23562. 

Need to use the latest build. You can try it at https://github.com/chrislusf/seaweedfs/releases/tag/dev

**Comment by chaos369** [comment]
After upgrade to weed-2021-03-04-22-05-linux-amd64.tar.gz, still can not work.

[signal SIGSEGV: segmentation violation code=0x1 addr=0x47 pc=0x7faa226495ca]

runtime stack:
runtime.throwI0305 08:18:07 08499 volume_loading.go:86] readSuperBlock volume 34 version 3
(0x243e744, 0x2a)
	/usr/local/go/src/runtime/panic.go:1112 +0x72
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:671 +I0305 08:18:07 08499 volume_loading.go:133] loading index /mnt/sdb/data/weed/file_info_34.idx to memory
0x46a

goroutine 123 [I0305 08:18:07 08499 disk_location.go:134] data file /mnt/sdb/data/weed/file_info_33.dat, replicaPlacement=000 v=3 size=32173104784 ttl=
syscall]:
runtime.cgocall(0x1c74860, 0xc00007b5c0, 0xc0005f0098)
	/usr/local/go/src/runtime/cgocall.go:133 +0x5b fp=0xc00007b590 sp=0xc00007b558 pc=0x40788b
net._C2func_getaddrinfo(0xc00039c840, 0x0, 0xc000617860, 0xc0005f0098, 0x0, 0x0, 0x0)
	_cgo_gotypes.go:92 +0x55 fp=0xc00007b5c0 sp=0xc00007b590 pc=0x5966b5
net.cgoLookupIPCNAME.func1I0305 08:18:07 08499 volume_loading.go:86] readSuperBlock volume 27 version 3
(0xc00039c840, 0x15, 0x15, 0xc000617860, 0xc0005f0098, 0x0, 0xc00007b6a0, 0x599a92)
	/usr/local/go/src/net/cgo_unix.go:161 +0xd2I0305 08:18:07 08499 volume_loading.go:133] loading index /mnt/sdb/data/weed/file_info_27.idx to memory
 fp=0xc00007b608 sp=0xc00007b5c0 pc=0x59c112
net.cgoLookupIPCNAME(0x23e4db1, 0x3, 0xc00039c760, 0x14, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...I0305 08:18:07 08499 volume_loading.go:86] readSuperBlock volume 39 version 3
)
	/usr/local/go/src/net/cgo_unix.go:161 +0x183 fp=0xc00007b718 sp=0xc00007b608 pc=0x597b43
net.cgoIPLookupI0305 08:18:07 08499 volume_loading.go:133] loading index /mnt/sdb/data/weed/file_info_39.idx to memory
(0xc0005c2a80, 0x23e4db1, 0x3, 0xc00039c760, I0305 08:18:07 08499 volume_loading.go:133] loading index /mnt/sdb/data/weed/file_info_38.idx to memory
0x14)
	/usr/local/go/src/net/cgo_unix.go:218 +0x67 fp=0xc00007b7b8 sp=0xc00007b718 pc=0x598277
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00007b7c0 sp=0xc00007b7b8 pc=0x46bd11
created by net.cgoLookupIP
	/usr/local/go/src/net/cgo_unix.go:228 +0xc7

goroutine 1 [select]:
net.(*Resolver).lookupIPAddr(0x3f1dca0, 0x29b72c0, 0xc000130020, 0x23e4db1, 0x3, 0xc00039c760, 0x14, 0x4a92, 0x0, 0x0, ...)
	/usr/local/go/src/net/lookup.go:274 +0x664
net.(*Resolver).internetAddrList(0x3f1dca0, 0x29b72c0, 0xc000130020, 0x23e4db1, 0x3, 0xc00039c760, 0x1a, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/ipsock.go:280 +0x4da
net.(*Resolver).resolveAddrList(0x3f1dca0, 0x29b72c0, 0xc000130020, 0x23e84be, 0x6, 0x23e4db1, 0x3, 0xc00039c760, 0x1a, 0x0, ...)
	/usr/local/go/src/net/dial.go:222 +0x49e
net.(*ListenConfig).Listen(0xc0007d7398, 0x29b72c0, 0xc000130020, 0x23e4db1, 0x3I0305 08:18:07 08499 disk_location.go:134] data file /mnt/sdb/data/weed/file_info_3.dat, replicaPlacement=000 v=3 size=31543971280 ttl=
, 0xc00039c760, 0x1a, 0x5, 0x1a, 0x5, ...)
	/usr/local/go/src/net/dial.go:624 +0xaa
net.Listen(0x23e4db1, 0x3, 0xc00039c760, 0x1a, 0x0, 0xc0007d7420, 0x3, 0x3)
	/usr/local/go/src/net/dial.go:707 +0x80
github.com/chrislusf/seaweedfs/weed/util.NewListener(0xc00039c760, 0x1a, 0x0, 0x23e42df, 0x1, 0xc0003b0eb0, 0x5)
	I0305 08:18:07 08499 volume_loading.go:86] readSuperBlock volume 40 version 3
/github/workspace/weed/util/net_timeout.go:82 +I0305 08:18:07 08499 disk_location.go:134] data file /mnt/sdb/data/weed/file_info_19.dat, replicaPlacement=000 v=3 size=31981391080 ttl=
0x4e
github.com/chrislusf/seaweedfs/weed/command.VolumeServerOptions.startGrpcService(0xc00069ab50, 0xc00069ab68, 0xc0003973d0, 0x1, 0x1, I0305 08:18:07 08499 volume_loading.go:133] loading index /mnt/sdb/data/weed/file_info_40.idx to memory
0xc00069b838, 0x1, 0x1, 0xc00071b850, 0xc00071b620, ...)
	/github/workspace/weed/command/volume.go:308 +0xc7
github.com/chrislusf/seaweedfs/weed/command.VolumeServerOptions.startVolumeServer(0xc00069ab50, 0xc00069ab68, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00071b850, I0305 08:18:07 08499 volume_loading.go:86] readSuperBlock volume 41 version 3
0xc00071b620, ...)
	/github/workspace/weed/command/volume.go:243I0305 08:18:07 08499 disk_location.go:134] data file /mnt/sdb/data/weed/file_info_20.dat, replicaPlacement=000 v=3 size=31504485088 ttl=
 +0xae3
I0305 08:18:07 08499 volume_loading.go:133] loading index /mnt/sdb/data/weed/file_info_41.idx to memory
github.com/chrislusf/seaweedfs/weed/command.runVolume(0x3e02380, 0xc000136080, 0x0, 0x0, 0x0)
	/github/workspace/weed/command/volume.go:122 +0x144
main.main()
	/github/workspace/weed/weed.go:66 +0x2ff

**Comment by chrislusf** [comment]
I reformatted the stack trace. The error happens when volume server is trying to listen to a port. You can try to remove the `-ip.bind` section.

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x47 pc=0x7faa226495ca]

runtime stack:
runtime.throw(0x243e744, 0x2a)
	/usr/local/go/src/runtime/panic.go:1112 +0x72
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:671 +0x46a

goroutine 123 [syscall]:
runtime.cgocall(0x1c74860, 0xc00007b5c0, 0xc0005f0098)
	/usr/local/go/src/runtime/cgocall.go:133 +0x5b fp=0xc00007b590 sp=0xc00007b558 pc=0x40788b
net._C2func_getaddrinfo(0xc00039c840, 0x0, 0xc000617860, 0xc0005f0098, 0x0, 0x0, 0x0)
	_cgo_gotypes.go:92 +0x55 fp=0xc00007b5c0 sp=0xc00007b590 pc=0x5966b5
net.cgoLookupIPCNAME.func1(0xc00039c840, 0x15, 0x15, 0xc000617860, 0xc0005f0098, 0x0, 0xc00007b6a0, 0x599a92)
	/usr/local/go/src/net/cgo_unix.go:161 +0xd2 fp=0xc00007b608 sp=0xc00007b5c0 pc=0x59c112
net.cgoLookupIPCNAME(0x23e4db1, 0x3, 0xc00039c760, 0x14, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/cgo_unix.go:161 +0x183 fp=0xc00007b718 sp=0xc00007b608 pc=0x597b43
net.cgoIPLookup(0xc0005c2a80, 0x23e4db1, 0x3, 0xc00039c760, 0x14)
	/usr/local/go/src/net/cgo_unix.go:218 +0x67 fp=0xc00007b7b8 sp=0xc00007b718 pc=0x598277
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc00007b7c0 sp=0xc00007b7b8 pc=0x46bd11
created by net.cgoLookupIP
	/usr/local/go/src/net/cgo_unix.go:228 +0xc7

goroutine 1 [select]:
net.(*Resolver).lookupIPAddr(0x3f1dca0, 0x29b72c0, 0xc000130020, 0x23e4db1, 0x3, 0xc00039c760, 0x14, 0x4a92, 0x0, 0x0, ...)
	/usr/local/go/src/net/lookup.go:274 +0x664
net.(*Resolver).internetAddrList(0x3f1dca0, 0x29b72c0, 0xc000130020, 0x23e4db1, 0x3, 0xc00039c760, 0x1a, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/ipsock.go:280 +0x4da
net.(*Resolver).resolveAddrList(0x3f1dca0, 0x29b72c0, 0xc000130020, 0x23e84be, 0x6, 0x23e4db1, 0x3, 0xc00039c760, 0x1a, 0x0, ...)
	/usr/local/go/src/net/dial.go:222 +0x49e
net.(*ListenConfig).Listen(0xc0007d7398, 0x29b72c0, 0xc000130020, 0x23e4db1, 0x3, 0xc00039c760, 0x1a, 0x5, 0x1a, 0x5, ...)
	/usr/local/go/src/net/dial.go:624 +0xaa
net.Listen(0x23e4db1, 0x3, 0xc00039c760, 0x1a, 0x0, 0xc0007d7420, 0x3, 0x3)
	/usr/local/go/src/net/dial.go:707 +0x80
github.com/chrislusf/seaweedfs/weed/util.NewListener(0xc00039c760, 0x1a, 0x0, 0x23e42df, 0x1, 0xc0003b0eb0, 0x5)
	/github/workspace/weed/util/net_timeout.go:82 +0x4e
github.com/chrislusf/seaweedfs/weed/command.VolumeServerOptions.startGrpcService(0xc00069ab50, 0xc00069ab68, 0xc0003973d0, 0x1, 0x1, 0xc00069b838, 0x1, 0x1, 0xc00071b850, 0xc00071b620, ...)
	/github/workspace/weed/command/volume.go:308 +0xc7
github.com/chrislusf/seaweedfs/weed/command.VolumeServerOptions.startVolumeServer(0xc00069ab50, 0xc00069ab68, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00071b850, 0xc00071b620, ...)
	/github/workspace/weed/command/volume.go:243 +0xae3
github.com/chrislusf/seaweedfs/weed/command.runVolume(0x3e02380, 0xc000136080, 0x0, 0x0, 0x0)
	/github/workspace/weed/command/volume.go:122 +0x144
main.main()
	/github/workspace/weed/weed.go:66 +0x2ff


```

**Comment by chrislusf** [comment]
> Thanks for your great job. It works well after upgrade from 2.17 to 2.31, but I found it prints the log below, not sure it effects:
> E0311 01:47:31 56950 filer_server_handlers_read.go:159] failed to write entry content:
> 

Not an error but a print bug. Fixed in 5d446673a6d4b7be49d0b2921b40894b760e7fe8

> And could tell me how to restart the service gracefully without kill -9 ? Thank you.

Use `kill` but not `kill -9`. The system is designed to handle non-graceful shutdown.
